File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed) =>
        speed switch
        {
            < 1 => 0.0,
            <= 4 => 1.0,
            <= 8 => 0.9,
            9 => 0.8,
            10 => 0.77,
            _ => 0.0
        };

    public static double ProductionRatePerHour(int speed) => SuccessRate(speed) * 221 * speed;

    public static int WorkingItemsPerMinute(int speed) => (int)(ProductionRatePerHour(speed) / 60);
}

File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int): float =
    if speed >= 10 then 0.77
    elif speed >= 9 then 0.8
    elif speed >= 5 then 0.9
    elif speed >= 1 then 1
    else 0

let productionRatePerHour (speed: int): float = float(speed * 221) * successRate speed

let workingItemsPerMinute (speed: int): int =
    productionRatePerHour speed / 60.0 |> floor |> int
